keel-bakery-plugin/src/main/kotlin/com/netflix/spinnaker/keel/bakery/artifact/ImageHandler.kt

```diff
@@ -122,7 +123,12 @@ class ImageHandler(
     diff: DefaultResourceDiff<Image>
   ): List<Task> {
     // TODO: Frigga and Rocket version parsing are not aligned. We should consolidate.
-    val appVersion = AppVersion.parseName(desiredVersion)
+    val appVersion = try {
+      AppVersion.parseName(desiredVersion)
+    } catch (ex: Exception) {
+      log.error("trying to parse desired version for artifact ${artifact.name} with version $desiredVersion but got an exception", ex)
+      throw SystemException("Invalid for artifact ${artifact.name} with version $desiredVersion")
```

Comment:
```suggestion
      throw SystemException("Invalid for artifact ${artifact.name} with version $desiredVersion", ex)
```

Reply:
If you pass the original exception as the second argument to `SystemException`, then its stack trace will also appear on the log (it will show the SystemException stack trace, then it will say "caused by..." and show the original stack trace). If you do that, I think you can eliminate the `log.error` statement on line 129.

Reply:
oh nice! let me do that.